### PR TITLE
fix panic in infohash.FromHexString

### PR DESCRIPTION
**Pull request by azr**
Howdy ! I managed to provoke this one somehow, so here's a fix.

### CODE REVIEW

File: types/infohash/infohash.go [hunk 1]
```diff
@@ -46,7 +46,8 @@ func (t *T) FromHexString(s string) (err error) {
 		return
 	}
 	if n != Size {
-		panic(n)
+		err = fmt.Errorf("decoded hash hex string has bad length: %d", n)
```
Comment (azr) [inline types/infohash/infohash.go:49]: probably initially gave the right length, but the string was garbage.

Comment (anacrolix) [reply types/infohash/infohash.go:49]: @azr what input string was able to cause the panic? I added a fuzz test to try to find anything and I'm not having any luck.

**Comment by anacrolix** [comment]
@azr did you have any insight into the comment I made?

File: types/infohash/infohash.go (hunk 1 above)
Comment (anacrolix) [reply types/infohash/infohash.go:49]: Bummer. I'll leave it as is, if you encounter the problem again, let me know.

**Pull request closed without merging**